selenium/pages/asset_library_page.rb

```diff
@@ -0,0 +1,110 @@
+require_relative '../spec/spec_helper'
+
+class AssetLibraryPage
+
+  include PageObject
+  include Logging
+
+  # SEARCH
+  text_area(:search_input, :id => 'assetlibrary-list-search')
+  button(:advanced_search_button, :xpath => '//button[@title="Advanced search"]')
+  button(:search_button, :xpath => '//button[@title="Search"]')
+
+  # ADD SITE
+  link(:add_site_link, :xpath => '//a[contains(.,"Add Site")]')
+  h2(:add_url_heading, :xpath => '//h2[text()="Add a URL"]')
+  text_area(:url_input, :id => 'assetlibrary-addlink-url')
+  text_area(:url_title_input, :id => 'assetlibrary-addlink-title')
+  select_list(:url_category, :id => 'assetlibrary-addlink-category')
+  text_area(:url_description, :id => 'assetlibrary-addlink-description')
+  link(:cancel_url_link, :text => 'Cancel')
+  button(:add_url_button, :xpath => '//button[text()="Add URL"]')
+  div(:missing_url_error, :xpath => '//div[text()="Please enter a URL"]')
+  div(:missing_url_title_error, :xpath => '//div[text()="Please enter a title"]')
+  div(:bad_url_error, :xpath => '//div[text()="Please enter a valid URL"]')
+
+  # MANAGE CATEGORIES
+  link(:manage_categories_link, :xpath => '//a[contains(.,"Manage categories")]')
+  h2(:manage_categories_heading, :xpath => '//h2[text()="Manage Categories"]')
+  text_area(:category_input, :id => 'assetlibrary-categories-create')
+  button(:add_category_button, :xpath => '//button[text()="Add"]')
+  elements(:category_title, :span, :xpath => '//span[@data-ng-bind="category.title"]')
+  elements(:edit_category_button, :button, :xpath => '//button[@title="Edit this category"]')
+  elements(:delete_category_button, :button, :xpath => '//button[@title="Delete this category"]')
+
+  # ASSETS
+  elements(:gallery_asset_link, :link, :xpath => '//li[@data-ng-repeat="asset in assets"]//a')
+  elements(:gallery_asset_title, :h3, :xpath => '//li[@data-ng-repeat="asset in assets"]//h3')
+  elements(:gallery_asset_owner_name, :element, :xpath => '//li[@data-ng-repeat="asset in assets"]//small')
+
+  def load_page(driver, url)
+    navigate_to url
+    wait_until(timeout=WebDriverUtils.page_load_wait) { driver.find_element(:id, 'tool_content') }
```

Comment:
Is this just a timeout after which it will check the existence of the element or will it wait for the element to become available? If there's a way to do the former approach, it would probably speed up the tests quite a bit.